app/search/dataone.py

```diff
@@ -147,11 +150,21 @@ def convert_result(self, result):
                             (boundingbox['east'], boundingbox['south']), ]
                     ]
                 )
+        # passing the dictionary with the original data sources
+        self.data_source_key =  result.pop('datasource', '').lstrip("urn:node:")
+        if self.data_source_key in app.datasources:
```

Comment:
It occurred to me after our conversation about how to do this that because you made the method itself cached, you can call `app.get_original_dataone_sources()` here and do away with the `app.datasources` and `app.gleaner_datasources` variables entirely. If you do that, then the first time you make a query will be slow, but the rest of them will be fast, because the datasources will be in the cache.

Reply:
i just changed it to app.get_original_dataone_sources() and i get this error that it does not exist. 
And i am pretty sure it cached
![image](https://user-images.githubusercontent.com/45835101/179301003-13b00daf-0396-4385-ae05-7bce84bbbc76.png)

Reply:
oh right - that makes sense, because the methods aren't defined in the`app` object. Sorry, in order to do it that way, we'd need to put them in a different file and import them from there.

I wanted to try this out, but I can't run it right now because there are some things missing from `requirements.txt`. What did you need to `pip install` in order to get this to work on your machine? If you add those to `requirements.txt`,  I'll be able to try it - and deployed environments will need that in order to work.